ext/bg/js/background-globals.js

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2020  Alex Yatskov <alex@foosoft.net>
+ * Author: Alex Yatskov <alex@foosoft.net>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+window.isBackground = true;
```

Comment:
Could probably save a file by putting this into core.js by doing something like:

```js
// In Yomichan constructor
this.isBackground = (
    typeof chrome.runtime.getBackgroundPage === 'function' &&
    chrome.runtime.getBackgroundPage() === window
);
```

I think the window references are the same, but I didn't test.

Reply:
I'm not sure how that would work on browser_action, have to test